tests/Feature/GraphQL/Mutations/CreateListingTest.php

```diff
@@ -374,79 +374,95 @@ public function test_it_will_fail_with_invalid_parameter_auction_data(): void
+                'type' => ListingType::AUCTION->name,
+                'auctionParams' => [
+                    'startBlock' => null,
+                    'endBlock' => fake()->numberBetween(1011, 2000),
+                ],
             ]]),
             true
         );
         $this->assertStringContainsString(
-            'Variable "$auctionData" got invalid value null at "auctionData.startBlock"; Expected non-nullable type "Int!" not to be null.',
+            'invalid value null at "listingData.auctionParams.startBlock"',
             $response['error']
         );
 
         $response = $this->graphql(
             $this->method,
-            array_merge($data, ['auctionData' => [
-                'startBlock' => Hex::MAX_UINT128 + 1,
-                'endBlock' => Hex::MAX_UINT128 + 1,
+            array_merge($data, ['listingData' => [
+                'type' => ListingType::AUCTION->name,
+                'auctionParams' => [
+                    'startBlock' => Hex::MAX_UINT128 + 1,
+                    'endBlock' => Hex::MAX_UINT128 + 1,
+                ],
             ]]),
             true
         );
         $this->assertStringContainsString(
-            'Variable "$auctionData" got invalid value 3.4028236692094E+38 at "auctionData.startBlock"; Int cannot represent non 32-bit signed integer value: 3.4028236692094E+38',
+            '"listingData.auctionParams.startBlock"; Int cannot represent non',
             $response['errors'][0]['message']
         );
         $this->assertStringContainsString(
-            'Variable "$auctionData" got invalid value 3.4028236692094E+38 at "auctionData.endBlock"; Int cannot represent non 32-bit signed integer value: 3.4028236692094E+38',
+            '"listingData.auctionParams.endBlock"; Int cannot represent non',
             $response['errors'][1]['message']
         );
 
         $response = $this->graphql(
             $this->method,
-            array_merge($data, ['auctionData' => [
-                'startBlock' => 1,
-                'endBlock' => 2,
+            array_merge($data, ['listingData' => [
+                'type' => ListingType::AUCTION->name,
+                'auctionParams' => [
+                    'startBlock' => 1,
+                    'endBlock' => 2,
+                ],
             ]]),
             true
         );
         $this->assertArraySubset(
-            ['auctionData.startBlock' => ['The auction data.start block must be at least 1010.']],
+            ['listingData.auctionParams.startBlock' => ['The listing data.auction params.start block must be at least 1010.']],
             $response['error']
         );
 
         $response = $this->graphql(
             $this->method,
-            array_merge($data, ['auctionData' => [
-                'startBlock' => 1012,
-                'endBlock' => 1011,
+            array_merge($data, ['listingData' => [
+                'type' => ListingType::AUCTION->name,
```

Comment:
Please could we add tests for the Offer type as well?

Reply:
Sure 😀